httpx/_content_streams.py

```diff
@@ -173,6 +173,35 @@ async def __aiter__(self) -> typing.AsyncIterator[bytes]:
         yield self.body
 
 
+class AsyncMultipartContent(ContentStream):
+    """
+    Async streaming multipart request content.
+    """
+
+    # NOTE: this helper is part of the public API.
+
+    def __init__(
+        self, aiterator: typing.AsyncIterator[bytes], content_length: int
+    ) -> None:
+        self._aiterator = aiterator
+        self._content_length = content_length
+
+    def can_replay(self) -> bool:
+        return False
+
+    def get_headers(self) -> typing.Dict[str, str]:
+        return {"Transfer-Encoding": "chunked"}
+
```

Comment:
I can probably drop these two + `__iter__()` and not make this derive from any base class - it's not actually used as a proper content stream in the multipart impl.